### DISCUSSION TITLE
homer - love it - but updates wipe out my config

### DISCUSSION THREAD

**Discussion by bingobob** [tteck/Proxmox#3020] [Help Needed] [answered]
Hi all

Title says it all, I update with the cli and type "update"

After that unfortunately it messes with my wipe my config. Any tips please?

Thanks

BB

**Comment by tteck** [comment]
It should backup your config
https://github.com/tteck/Proxmox/blob/2a8a65e7544e8a9cbfacaf5246d4f643e798b8db/ct/homer.sh#L62-L66

and restore it
https://github.com/tteck/Proxmox/blob/2a8a65e7544e8a9cbfacaf5246d4f643e798b8db/ct/homer.sh#L74-L78

If it's not, I'll look into it when I find time.

**Reply by bingobob** [reply]
Thanks, wondering if it's because I have more than one config. Or if it's that I've downloaded loads of custom icons. Or maybe the update script in using isn't the latest. Not sure. Good that it's meant to cope with keep custom config though.

**Reply by liveware-problems** [reply]
If you have additional configs in `/assets/*.yml` files, they will be wiped by the update, along with anything you add to the `/assets/icons/` directory. I have persisted my icons by saving them to the `/assets/tools/` directory and referencing them from tools as this folder is persisted when updating, as @tteck has detailed above.

**Reply by liveware-problems** [reply, marked as answer]
@bingobob the update script will now backup and restore the entire `/assets/` directory - working fine for my additional page.yml files and icons now 👍

**Reply by bingobob** [reply]
Thanks guys, working fine now (just updated).

**Comment by bingobob** [comment]
Slightly related question. In some of my tteck LXCs I just type "update" to update, in others, apt update && apt upgrade and in others like this homer one, i rerun the script and select update.

bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/homer.sh)"


am i doing it right ?

thanks

**Reply by tteck** [reply]
yes

**Reply by bingobob** [reply]
Good! Thanks.
